lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystem.java

```diff
@@ -388,4 +391,13 @@ private static void validateS3Location(Location location)
     {
         new S3Location(location);
     }
+
+    // TODO: remove me once all of the S3-compatible storage supports new integrity checks
```

Comment:
how do we know when is that. DOes not seem actionable

Reply:
Once the Vast and ECS are there. ECS isn't (I've just tested) and Vast reported the same on the Slack.